File: src/utils/languageConfig.ts
/**
 * AI Writer — Language Configuration
 *
 * Contains the list of supported languages and their metadata.
 * Supports auto-detection of device language via expo-localization.
 */

import * as Localization from 'expo-localization';

export interface LanguageOption {
  code: string;
  name: string;
  nativeName: string;
  direction: 'ltr' | 'rtl';
}

export const SUPPORTED_LANGUAGES: LanguageOption[] = [
  { code: 'en', name: 'English', nativeName: 'English', direction: 'ltr' },
  { code: 'es', name: 'Spanish', nativeName: 'Español', direction: 'ltr' },
  { code: 'fr', name: 'French', nativeName: 'Français', direction: 'ltr' },
  { code: 'de', name: 'German', nativeName: 'Deutsch', direction: 'ltr' },
  { code: 'zh', name: 'Chinese', nativeName: '中文', direction: 'ltr' },
  { code: 'ja', name: 'Japanese', nativeName: '日本語', direction: 'ltr' },
  { code: 'ko', name: 'Korean', nativeName: '한국어', direction: 'ltr' },
  { code: 'ar', name: 'Arabic', nativeName: 'العربية', direction: 'rtl' },
  { code: 'ru', name: 'Russian', nativeName: 'Русский', direction: 'ltr' },
  { code: 'hi', name: 'Hindi', nativeName: 'हिन्दी', direction: 'ltr' },
  { code: 'pt', name: 'Portuguese', nativeName: 'Português', direction: 'ltr' },
  { code: 'it', name: 'Italian', nativeName: 'Italiano', direction: 'ltr' },
  { code: 'tr', name: 'Turkish', nativeName: 'Türkçe', direction: 'ltr' },
  { code: 'nl', name: 'Dutch', nativeName: 'Nederlands', direction: 'ltr' },
  { code: 'pl', name: 'Polish', nativeName: 'Polski', direction: 'ltr' },
  { code: 'sv', name: 'Swedish', nativeName: 'Svenska', direction: 'ltr' },
  { code: 'th', name: 'Thai', nativeName: 'ไทย', direction: 'ltr' },
  { code: 'vi', name: 'Vietnamese', nativeName: 'Tiếng Việt', direction: 'ltr' },
  { code: 'id', name: 'Indonesian', nativeName: 'Bahasa Indonesia', direction: 'ltr' },
  { code: 'he', name: 'Hebrew', nativeName: 'עברית', direction: 'rtl' },
  { code: 'ms', name: 'Malay', nativeName: 'Bahasa Melayu', direction: 'ltr' },
  { code: 'bn', name: 'Bengali', nativeName: 'বাংলা', direction: 'ltr' },
  { code: 'ur', name: 'Urdu', nativeName: 'اردو', direction: 'rtl' },
  { code: 'fa', name: 'Persian', nativeName: 'فارسی', direction: 'rtl' },
  { code: 'uk', name: 'Ukrainian', nativeName: 'Українська', direction: 'ltr' },
  { code: 'ro', name: 'Romanian', nativeName: 'Română', direction: 'ltr' },
  { code: 'cs', name: 'Czech', nativeName: 'Čeština', direction: 'ltr' },
  { code: 'el', name: 'Greek', nativeName: 'Ελληνικά', direction: 'ltr' },
  { code: 'hu', name: 'Hungarian', nativeName: 'Magyar', direction: 'ltr' },
  { code: 'fi', name: 'Finnish', nativeName: 'Suomi', direction: 'ltr' },
  { code: 'da', name: 'Danish', nativeName: 'Dansk', direction: 'ltr' },
  { code: 'no', name: 'Norwegian', nativeName: 'Norsk', direction: 'ltr' },
];

/**
 * Get a language by its code.
 */
export function getLanguageByCode(code: string): LanguageOption | undefined {
  return SUPPORTED_LANGUAGES.find((lang) => lang.code === code);
}

/**
 * Detect the device language and return the matching supported language.
 * Falls back to English if the device language is not supported.
 */
export function detectDeviceLanguage(): LanguageOption {
  try {
    // expo-localization provides an array of locale strings
    const locales = Localization.getLocales();
    if (locales && locales.length > 0) {
      const deviceLangCode = locales[0].languageCode;
      if (deviceLangCode) {
        const match = getLanguageByCode(deviceLangCode);
        if (match) return match;
      }
    }
  } catch (e) {
    console.warn('Failed to detect device language:', e);
  }

  // Fallback to English
  return SUPPORTED_LANGUAGES[0];
}

/**
 * Default language — auto-detected from device.
 */
export const DEFAULT_LANGUAGE = detectDeviceLanguage();

File: app/src/main/java/com/smartaiwriter/app/util/LanguageConfig.kt
package com.smartaiwriter.app.util

data class LanguageItem(
    val code: String,
    val name: String,
    val nativeName: String,
    val isRtl: Boolean = false
)

object LanguageConfig {
    /** Languages available for content generation (32) */
    val contentLanguages = listOf(
        LanguageItem("en", "English", "English"),
        LanguageItem("es", "Spanish", "Español"),
        LanguageItem("fr", "French", "Français"),
        LanguageItem("de", "German", "Deutsch"),
        LanguageItem("zh", "Chinese", "中文"),
        LanguageItem("ja", "Japanese", "日本語"),
        LanguageItem("ko", "Korean", "한국어"),
        LanguageItem("ar", "Arabic", "العربية", isRtl = true),
        LanguageItem("ru", "Russian", "Русский"),
        LanguageItem("hi", "Hindi", "हिन्दी"),
        LanguageItem("pt", "Portuguese", "Português"),
        LanguageItem("it", "Italian", "Italiano"),
        LanguageItem("tr", "Turkish", "Türkçe"),
        LanguageItem("nl", "Dutch", "Nederlands"),
        LanguageItem("pl", "Polish", "Polski"),
        LanguageItem("sv", "Swedish", "Svenska"),
        LanguageItem("th", "Thai", "ไทย"),
        LanguageItem("vi", "Vietnamese", "Tiếng Việt"),
        LanguageItem("id", "Indonesian", "Bahasa Indonesia"),
        LanguageItem("he", "Hebrew", "עברית", isRtl = true),
        LanguageItem("ms", "Malay", "Bahasa Melayu"),
        LanguageItem("bn", "Bengali", "বাংলা"),
        LanguageItem("ur", "Urdu", "اردو", isRtl = true),
        LanguageItem("fa", "Persian", "فارسی", isRtl = true),
        LanguageItem("uk", "Ukrainian", "Українська"),
        LanguageItem("ro", "Romanian", "Română"),
        LanguageItem("cs", "Czech", "Čeština"),
        LanguageItem("el", "Greek", "Ελληνικά"),
        LanguageItem("hu", "Hungarian", "Magyar"),
        LanguageItem("fi", "Finnish", "Suomi"),
        LanguageItem("da", "Danish", "Dansk"),
        LanguageItem("no", "Norwegian", "Norsk")
    )

    /** Languages available for app UI (15) */
    val appLanguages = listOf(
        LanguageItem("en", "English", "English"),
        LanguageItem("ar", "Arabic", "العربية", isRtl = true),
        LanguageItem("es", "Spanish", "Español"),
        LanguageItem("fr", "French", "Français"),
        LanguageItem("de", "German", "Deutsch"),
        LanguageItem("zh", "Chinese", "中文"),
        LanguageItem("ja", "Japanese", "日本語"),
        LanguageItem("ko", "Korean", "한국어"),
        LanguageItem("ru", "Russian", "Русский"),
        LanguageItem("hi", "Hindi", "हिन्दी"),
        LanguageItem("pt", "Portuguese", "Português"),
        LanguageItem("tr", "Turkish", "Türkçe"),
        LanguageItem("it", "Italian", "Italiano"),
        LanguageItem("fa", "Persian", "فارسی", isRtl = true),
        LanguageItem("ur", "Urdu", "اردو", isRtl = true)
    )

    fun isRtlLanguage(code: String): Boolean {
        return contentLanguages.find { it.code == code }?.isRtl == true
    }

    fun getLanguageName(code: String): String {
        return contentLanguages.find { it.code == code }?.name ?: code
    }
}
